gufe/custom_json.py

```diff
@@ -0,0 +1,162 @@
+# This file was originally part of OpenPathSampling/SimStore, which is
+# distributed under the MIT license.
+# Portions Copyright (c) 2014-2022 the contributors to OpenPathSampling
+# Permissions are the same as those listed in the gufe LICENSE
+
+import json
+import functools
+
+
+class JSONSerializerDeserializer(object):
```

Comment:
type annotations on this would help a lot

Reply:
Added in [c7abb1c](https://github.com/OpenFreeEnergy/gufe/pull/68/commits/c7abb1c7c17abb5ae5e8baa67ec3124300377e66), although I'm not sure how much they actually help....